GPT/beta-commands/gpt_callables.py

```diff
@@ -0,0 +1,118 @@
+def _generate_tool(func: Callable):
+    sig = inspect.signature(func)
+    schema = {
+        "type": "function",
+        "function": {
+            "name": func.__name__,
+            "description": inspect.getdoc(func),
+            "parameters": {
+                "type": "object",
+                "properties": {},
+                "required": [],
+            },
+        },
+    }
+
+    for name, param in sig.parameters.items():
+        if param.default is param.empty:
+            schema["function"]["parameters"]["required"].append(name)
+        schema["function"]["parameters"]["properties"][name] = {
+            "type": str(param.annotation).split("'")[1],
+            "description": "",
+        }
+
+    return schema
+
+def notify_user(response: str):
+    """Send a notification to the desktop"""
+    actions.app.notify(response)
+
+def search_for_command(response: str):
+    """Search VSCode for command"""
+    actions.user.command_palette()
+    actions.user.paste(response)
+
+def display_response(response: str):
+    """Open the GPT help file in the web browser"""
+    builder = Builder()
+    builder.h1("Displaying the Model Response")
+    builder.p(response)
+    builder.render()
+
+
+function_specs = [
+            {
+                "type": "function",
+                "function": {
+                    "name": "insert",
+                    "description": "Insert the string into the document. The document is in the language specified so if you aren't careful you will cause syntax errors.",
+                    "parameters": {
+                        "type": "object",
+                        "properties": {
+                            "str": {
+                                "type": "string",
+                                "description": "The text to insert",
+                            }
+                        },
+                        "required": ["str"],
+                    },
+                },
+            },
+            {
+                "type": "function",
+                "function": {
+                    "name": "notify",
+                    "description": "Notify the user using a popup notification",
+                    "parameters": {
+                        "type": "object",
+                        "properties": {
+                            "str": {
+                                "type": "string",
+                                "description": "The text to notify",
+                            }
+                        },
+                        "required": ["str"],
+                    },
+                },
+            },
+            {
+                "type": "function",
+                "function": {
+                    "name": "display",
+                    "description": "DEFAULT - Display the response to the user. Use this for all informational text aside from notifications. Use this instead of returning content in the response.",
```

Comment:
Is there a reason we specify DEFAULT? Is it just a heuristic? I can't find this or anything similar in the openai docs

Reply:
I was trying to get the prompt to trigger the display without falling back to displaying the content manually. I think we could remove it if you like as it's not clear to me that it worked. I copied the example from another OSS project.

Reply:
Ok I think we can remove it